src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static final User userNull = null;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    public void registerAndGetNewUser_Ok() {
+        User userFirst = createUser(22, "crimeabridgedestroyed", "Himars");
+        User userSecond = createUser(70, "freedom", "Bayraktar");
+        registrationService.register(userFirst);
+        User expectedUser1 = storageDao.get(Storage.people.get(0).getLogin());
+        assertEquals(userFirst, expectedUser1);
+        registrationService.register(userSecond);
+        User expectedUser2 = storageDao.get(Storage.people.get(1).getLogin());
+        assertEquals(userSecond, expectedUser2);
+    }
+
+    @Test
+    void registerAndGetDifferentUsers_Ok() {
```

Comment:
due to mentors comment, there is no need to create test case for 2 different users

Reply:
Why is it so? There might be a situation when the second user overwrites the first one, for example. I want to test all the possible mistakes